src/adaptations/device-layer/FreeRTOS/SystemTimeSupport.cpp

```diff
@@ -40,7 +40,7 @@ constexpr uint32_t kTicksOverflowShift = (configUSE_16_BIT_TICKS) ? 16 : 32;
 
 uint64_t sBootTimeUS = 0;
 
-uint64_t TicksSinceBoot(void)
+uint64_t TicksSinceBoot(void) __attribute__((weak))
```

Comment:
This isn't useful as-is because TicksSinceBoot() is declared in an anonymous namespace, so it can't be overridden. 

I suggest moving it into the Platform::Layer namespace and then renaming it something more specific, like FreeRTOSTicksSinceBoot().

Also I suggest adding documentation explaining the issue.  E.g.:

```
/**
 * Returns the number of FreeRTOS ticks since the system booted.
 *
 * NOTE: The default implementation of this function uses FreeRTOS's vTaskSetTimeOutState()
 * function to get the total number of ticks, irrespective of tick counter overflows.
 * Unfortunately, this function cannot be called in interrupt context, and no equivalent
 * ISR function exists. Therefore, adaptations that need to use the Weave Get/SetClock
 * methods from within an interrupt handler must override this function with a suitable
 * alternative that works on the target platform.
 */
uint64_t FreeRTOSTicksSinceBoot(void) __attribute__((weak))
```

Reply:
SGTM, will do.